PURL-SPECIFICATION.rst

```diff
@@ -215,37 +215,51 @@ The rules for each component are:
 Character encoding
 ~~~~~~~~~~~~~~~~~~
 
-For clarity and simplicity a ``purl`` is always an ASCII string. To ensure that
-there is no ambiguity when parsing a ``purl``, separator characters and non-ASCII
-characters must be UTF-encoded and then percent-encoded as defined at::
+Permitted characters
+--------------------
 
-    https://en.wikipedia.org/wiki/Percent-encoding
+A canonical ``purl`` is an ASCII string composed of these characters:
 
-Use these rules for percent-encoding and decoding ``purl`` components:
+- alphanumeric characters ``A to Z``, ``a to z``, ``0 to 9``,
+- the ``purl`` separators ``:/@?=&#`` (colon ':', slash '/', at sign '@',
+  question mark '?', equal sign '=', ampersand '&' and pound sign '#'), and
+- these punctuation marks ``%.-_~`` (percent sign '%', period '.', dash '-',
+  underscore '_' and tilde '~').
 
-- the ``type`` must NOT be encoded and must NOT contain separators
+All other characters MUST be encoded as UTF-8 and then percent-encoded.
+In addition, each component specifies its permitted characters and
+its percent-encoding rules.
```

Comment:
Should we specify a [Unicode normal form](https://en.wikipedia.org/wiki/Unicode_equivalence#Normal_forms) to apply to components before UTF-8 encoding?

Certainly we should add a note about [overlong UTF-8 encodings](https://en.wikipedia.org/wiki/UTF-8#Overlong_encodings) (like `%C0%AE` to represent a dot `.`) to the parser, so that some validations are performed on the **decoded** form of components.

Reply:
UTF-8 is enough, no need for other complexity